### PR TITLE
Add BSD-2 license

### PR DESCRIPTION
**Pull request by sbesson**
As we are starting to publish this repository, I reviewed the licensing. https://github.com/ome/ZarrReader/blob/4558c079ff1ae23b38b1c53156e11266bab17eb1/src/loci/formats/in/ZarrReader.java#L2-L24 is currently GPL-2 but given the description it looks like it came from a copy-n-paste of existing reader/template rather than a decision.

Alongside our recent policy for readers/writers associated with open specification (see https://github.com/ome/bioformats/pull/3700 and https://github.com/ome/bioformats/pull/3712), this proposes to distribute this reader under the permissive BSD-2 license. Included are the minimal changes to the top-level POM and the outcome of the execution of `mvn license:update-file-header`

### CODE REVIEW

**Review by joshmoore — APPROVED** [review]
MSTM.

**Comment by sbesson** [comment]
The only thing I am unsure about is really the project name. But as the scope of this repository evolves, I assume we can easily review what's proposed here /cc @dgault

File: pom.xml [hunk 1]
```diff
@@ -157,6 +173,14 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>license-maven-plugin</artifactId>
+        <configuration>
+          <licenseName>bsd_2</licenseName>
+          <projectName>OME-Zarr Bio-Formats reader</projectName>
```
Comment (joshmoore) [inline pom.xml:181]: Based on https://github.com/ome/bioformats/blob/develop/components/formats-bsd/pom.xml#L233 perhaps something like:

- Zarr-based implementations of Bio-Formats readers and writers

Comment (joshmoore) [reply pom.xml:181]: or if the repo tends towards formats-ngff:

 * Next-generation implementations ...

Comment (joshmoore) [reply pom.xml]: k

**Pull request merged**